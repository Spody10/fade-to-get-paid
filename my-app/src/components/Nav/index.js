import React from 'react';
import roostyImage from "../../assets/roosty.jpg";

function Nav() {
    const categories = [
        {
            name: "Sports",
            description: "Sports available to pick"
        },
        {
            name: "Current Picks",
            description: "Current picks from the WOAT",
        },
        {
            name: "Previous Picks",
            description: "God he is awful",
        },

    ]
    function categorySelected(name) {
        console.log(`${name} clicked`)
    }
    return (
        <header>
            <h2>
                <a href="/">
                    <span role="img" aria-label="roosty">
                        <img src={roostyImage} className="my-2" style={{ width: "5%" }} alt =""></img>
                    </span> 
                </a>
            </h2>
            <nav>
                <ul className="flex-row">
                    <li className="mx-2">
                        <a href="#about">
                            About McNoosty
                        </a>
                    </li>
                    <li>
                        <span>Contact</span>
                    </li>
                    {categories.map((category) => (
                    <li 
                        className="mx-1"
                        key={category.name}
                    >
                        <span onClick={() => categorySelected(category.name)} >
                            {category.name}
                        </span>
                    </li>
                    ))}
                </ul>
            </nav>
        </header>
    );
}

export default Nav;