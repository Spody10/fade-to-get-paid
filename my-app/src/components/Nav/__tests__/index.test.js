import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
    //baseline test 
    it('renders', () => {
        render(<Nav />);
    });

    //snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);
        //assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('links are visable', () => {
    it('inserts text into the links', () => {
        //Arrange
        const { getByTestId } = render(<Nav />);
        
        //Assert 
        expect(getByTestId('link')).toHaveTextContent('Roosty');
        expect(getByTestId('about')).toHaveTextContent('About McNoosty')

    });
})