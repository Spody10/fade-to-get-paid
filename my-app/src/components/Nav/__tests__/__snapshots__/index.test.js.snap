// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        href="/"
      >
        <span
          aria-label="roosty"
          role="img"
        >
          <img
            alt=""
            class="my-2"
            src="roosty.jpg"
            style="width: 5%;"
          />
        </span>
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            href="#about"
          >
            About McNoosty
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Sports
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Current Picks
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Previous Picks
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
